load/load.go

```diff
@@ -2,12 +2,56 @@ package load
 
 import (
 	"context"
+	"fmt"
 
 	"github.com/hashicorp/consul/api"
 	"github.com/mkeeler/consul-load-test/metrics"
 )
 
-func Load(ctx context.Context, client *api.Client, conf Config, metricsServer *metrics.MetricsServer) {
-	go kvLoad(ctx, client, conf, metricsServer)
-	<-ctx.Done()
+func Load(ctx context.Context, client *api.Client, conf Config, metricsServer *metrics.MetricsServer) <-chan struct{} {
+	done := make(chan struct{})
+
+	go func() {
+		defer close(done)
+
+		var loadDone <-chan struct{}
+		loadType := getLoadType(conf)
```

Comment:
I don't quite follow why we need this extra method here. Could we instead just write a `if..else if...else` and avoid the indirection?

Reply:
Will update